tests/integrations/base/test_tool_registry.py

```diff
@@ -19,23 +18,23 @@ def integrations_modules() -> Set[str]:
     }
 
 
-def tools_for_module(tool_module_name: str) -> Dict[str, Type[Tool]]:
+def tools_for_module(tool_module_name: str) -> dict[str, type[Tool]]:
     """Return classes that subclass Tool in a module in ``briefcase.integrations``, e.g.
     {"android_sdk": AndroidSDK}."""
     return dict(
         inspect.getmembers(
             sys.modules[f"briefcase.integrations.{tool_module_name}"],
             lambda klass: (
                 inspect.isclass(klass)
+                and not inspect.isabstract(klass)
                 and issubclass(klass, (Tool, ManagedTool))
```

Comment:
This is the only substantive logic change required by the version upgrade. 

With the move of Sequence from `typing.Sequence` to `collections.abc.Sequence`, `SubprocessArgsT` now returns True from `inspect.isclass()`, but raises an error if you try to use it in `issubclass`. Adding an additional check for `inspect.isabstract()` avoids the error, and also allows removing the explicit reference to `Tool` and `ManagedTool` classes.